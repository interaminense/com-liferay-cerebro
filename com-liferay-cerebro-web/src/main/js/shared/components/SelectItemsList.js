import autobind from 'autobind-decorator';
import Component, {Config} from 'metal-jsx';
import {Map} from 'immutable';

import ListGroup from 'shared/components/list-group';
import Checkbox from 'shared/components/Checkbox';

class SelectItem extends Component {
	@autobind
	handleClick() {
		const {checked, id, onSelect} = this.props;

		onSelect(id, !checked);
	}

	render() {
		const {checked, id, label} = this.props;

		return (
			<ListGroup.Item
				elementClasses={checked ? 'active' : ''}
				flex
				key={id}
				onClick={this.handleClick}
			>
				<ListGroup.ItemField>
					<Checkbox checked={checked} />
				</ListGroup.ItemField>

				<ListGroup.ItemField expand>{label}</ListGroup.ItemField>
			</ListGroup.Item>
		);
	}
}

SelectItem.PROPS = {
	checked: Config.bool(),
	id: Config.oneOfType([Config.string(), Config.number()]),
	label: Config.string(),
	onSelect: Config.func()
};

class SelectItemsList extends Component {
	render() {
		const {selectedItemsIMap, items, onSelect} = this.props;

		return (
			<ListGroup>
				{items.map(({id, name}) => (
					<SelectItem
						checked={selectedItemsIMap.get(id)}
						id={id}
						key={id}
						label={name}
						onSelect={onSelect}
					/>
				))}
			</ListGroup>
		);
	}
}

SelectItemsList.PROPS = {
	items: Config.array(),
	onSelect: Config.func(),
	selectedItemsIMap: Config.instanceOf(Map).value(Map())
};

export default SelectItemsList;