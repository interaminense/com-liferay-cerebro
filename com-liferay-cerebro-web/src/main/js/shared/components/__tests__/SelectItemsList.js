import SelectItemsList from '../SelectItemsList';
import {Map} from 'immutable';

const ITEMS = [
	{
		id: 1,
		name: 'Portland'
	},
	{
		id: 2,
		name: 'San Diego'
	},
	{
		id: 3,
		name: 'Seattle'
	}
];

const SELECTED_ITEMS = Map({
	1: true,
	2: true,
	3: true
});

describe('SelectItemsList', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new SelectItemsList({items: ITEMS});

		expect(component).toMatchSnapshot();
	});

	it('should render with checked items', () => {
		component = new SelectItemsList({
			items: ITEMS,
			selectedItemsIMap: SELECTED_ITEMS
		});

		expect(component).toMatchSnapshot();
	});
});