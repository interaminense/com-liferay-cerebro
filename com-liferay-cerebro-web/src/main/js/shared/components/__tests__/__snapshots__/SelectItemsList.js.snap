// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectItemsList should render 1`] = `
<ul class="list-group list-group-root">
  <li class="list-group-item list-group-item-flex">
    <div class="flex-col">
      <div class="custom-control custom-checkbox">
        <label>
          <input class="custom-control-input" ref="checkbox" type="checkbox">
          <span class="custom-control-indicator"></span>
          <span class="custom-control-description"></span>
        </label>
      </div>
    </div>
    <div class="flex-col flex-col-expand">Portland</div>
  </li>
  <li class="list-group-item list-group-item-flex">
    <div class="flex-col">
      <div class="custom-control custom-checkbox">
        <label>
          <input class="custom-control-input" ref="checkbox" type="checkbox">
          <span class="custom-control-indicator"></span>
          <span class="custom-control-description"></span>
        </label>
      </div>
    </div>
    <div class="flex-col flex-col-expand">San Diego</div>
  </li>
  <li class="list-group-item list-group-item-flex">
    <div class="flex-col">
      <div class="custom-control custom-checkbox">
        <label>
          <input class="custom-control-input" ref="checkbox" type="checkbox">
          <span class="custom-control-indicator"></span>
          <span class="custom-control-description"></span>
        </label>
      </div>
    </div>
    <div class="flex-col flex-col-expand">Seattle</div>
  </li>
</ul>
`;

exports[`SelectItemsList should render with checked items 1`] = `
<ul class="list-group list-group-root">
  <li class="list-group-item list-group-item-flex">
    <div class="flex-col">
      <div class="custom-control custom-checkbox">
        <label>
          <input class="custom-control-input" ref="checkbox" type="checkbox">
          <span class="custom-control-indicator"></span>
          <span class="custom-control-description"></span>
        </label>
      </div>
    </div>
    <div class="flex-col flex-col-expand">Portland</div>
  </li>
  <li class="list-group-item list-group-item-flex">
    <div class="flex-col">
      <div class="custom-control custom-checkbox">
        <label>
          <input class="custom-control-input" ref="checkbox" type="checkbox">
          <span class="custom-control-indicator"></span>
          <span class="custom-control-description"></span>
        </label>
      </div>
    </div>
    <div class="flex-col flex-col-expand">San Diego</div>
  </li>
  <li class="list-group-item list-group-item-flex">
    <div class="flex-col">
      <div class="custom-control custom-checkbox">
        <label>
          <input class="custom-control-input" ref="checkbox" type="checkbox">
          <span class="custom-control-indicator"></span>
          <span class="custom-control-description"></span>
        </label>
      </div>
    </div>
    <div class="flex-col flex-col-expand">Seattle</div>
  </li>
</ul>
`;
